### PR TITLE
feat(alias)!: resolve entry id

### LINKED ISSUE #1190: Cannot use an alias as an `input`
- Rollup Plugin Name: @rollup/plugin-alias
- Rollup Plugin Version: 3.1.9
- Rollup Version: 2.74.0
- Operating System (or Browser): OSX
- Node Version: 16
- Link to reproduction _(⚠️ read below)_: https://replit.com/@matthewp17/rollup-plugin-alias-bug

### Expected Behavior

You should be able to pass an alias as an `input`.

### Actual Behavior

An error for not being able to resolve the alias is shown in the console.

### Additional Information

This is the code that causes the bug: https://github.com/rollup/plugins/blob/36de7cc9c3915667bc0dd203ee7b139da8858959/packages/alias/src/index.ts#L79-L81

Was introduced in this PR: https://github.com/rollup/plugins/pull/1024

Doesn't seem like the PR was intentionally trying to make an alias as an input not be allowed, so I think this part was a mistake.

### PR DESCRIPTION
**Pull request by bluwy**
## Rollup Plugin Name: `alias`

This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [x] yes (_breaking changes will not be merged unless absolutely necessary_)
- [ ] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. - Added

List any relevant issue numbers:
fix #1190

### Description

When the Rollup input is passed an aliased path, the `alias` plugin doesn't resolve it because it skips if there's no `importer`. I tracked this behaviour, and found it was not done in the past due to https://github.com/rollup/rollup-plugin-alias/pull/29, but since the internals changed now, that PR's issue isn't relevant and we can continue to alias entry paths now.

### CODE REVIEW

**Comment by bluwy** [comment]
My bad. Didn't notice my syntax in the other PR would also close this. Re-opened and it should still be good to go.

**Comment by shellscape** [comment]
> but since the internals changed now

could you link to the internals you mentioned?

**Comment by bluwy** [comment]
The old plugin code used to rely on the `importer` at https://github.com/rollup/rollup-plugin-alias/blob/0204677c0e1112cba7a5a0f60204ff1df81493b0/src/index.js#L80. Now we're not doing that anymore (we don't use the `path` nodejs module either). We're only passing the `importer` to the `customResolver`, which is currently correctly typed that `importer` could be defined.

**Review by shellscape — APPROVED** [review]
Makes sense, thanks. I'd like to get another approval from another maintainer before we merge.

**Pull request merged**
